services/ui-src/src/utils/api/requestMethods/report.ts

```diff
@@ -1,85 +1,90 @@
-import { API } from "aws-amplify";
+import { get, post, put } from "aws-amplify/api";
 import { ReportKeys, ReportShape } from "types";
 import { getRequestHeaders } from "./getRequestHeaders";
 import { updateTimeout } from "utils";
 
+const apiName = "mcr";
+
 async function archiveReport(reportKeys: ReportKeys) {
   const requestHeaders = await getRequestHeaders();
-  const request = {
+  const options = {
     headers: { ...requestHeaders },
   };
   const { reportType, state, id } = reportKeys;
+  const path = `/reports/archive/${reportType}/${state}/${id}`;
 
   updateTimeout();
-  const response = await API.put(
-    "mcr",
-    `/reports/archive/${reportType}/${state}/${id}`,
-    request
-  );
-  return response;
+  await put({
+    apiName,
+    path,
+    options,
+  }).response;
 }
 
 async function releaseReport(reportKeys: ReportKeys) {
   const requestHeaders = await getRequestHeaders();
-  const request = {
+  const options = {
     headers: { ...requestHeaders },
   };
   const { reportType, state, id } = reportKeys;
+  const path = `/reports/release/${reportType}/${state}/${id}`;
 
   updateTimeout();
-  const response = await API.put(
-    "mcr",
-    `/reports/release/${reportType}/${state}/${id}`,
-    request
-  );
-  return response;
+  await put({
+    apiName,
+    path,
+    options,
+  }).response;
 }
 
 async function submitReport(reportKeys: ReportKeys) {
   const requestHeaders = await getRequestHeaders();
-  const request = {
+  const options = {
     headers: { ...requestHeaders },
   };
   const { reportType, state, id } = reportKeys;
+  const path = `/reports/submit/${reportType}/${state}/${id}`;
 
   updateTimeout();
-  const response = await API.post(
-    "mcr",
-    `/reports/submit/${reportType}/${state}/${id}`,
-    request
-  );
-  return response;
+  const { body } = await post({
+    apiName,
+    path,
+    options,
+  }).response;
+  return (await body.json()) as unknown as ReportShape;
```

Comment:
Same question about `unknown`. I'm guessing that the casting makes more sense where these endpoints are actually called? But within these methods it isn't immediately obvious why the casting needs to happen.

Reply:
the [build was failing](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/actions/runs/11280305608/job/31373091951) previously so Ben pushed [this change](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11876/commits/5ab08c6b4b9c9f0eb3d6e321cf6d6e3177a6636e) to resolve it. I too don’t love the casting but I need the build to pass. Open to alternatives!